### PR TITLE
restart Dev Guide

### PR DESCRIPTION
**Pull request by eed3si9n**
/review @jsuereth

### CODE REVIEW

File: src/reference/03-Developers-Guide/10-DevGuide-Notes/00.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+out: DevGuide-Notes.html
+---
+
+Notes
```
Comment (jsuereth) [inline src/reference/03-Developers-Guide/10-DevGuide-Notes/00.md]: This was really about the architectural core of sbt.  I would like to keep it around/live.

File: src/reference/03-Developers-Guide/00.md [hunk 2]
```diff
@@ -2,12 +2,93 @@
...
-understand the fundamentals of `Setting[_]`, `Task[_]` and constructing
-builds.
+  [pickling]: https://github.com/scala/pickling
+  [HowCanIHelp]: http://www.scala-sbt.org/community.html#how-can-I-help
+  [Sbt-Launcher]: Sbt-Launcher.html
+
+Developer's Guide (Work in progress)
+------------------------------------
+
+This is the set of documentation about the Architecture of sbt.
```
Comment (jsuereth) [inline src/reference/03-Developers-Guide/00.md]: MOdularization != Architecture.   Modularization is more of a "core principle" or "goal".  Archtiecture is the actual structure of each module.

File: src/reference/03-Developers-Guide/00.md [hunk 3]
```diff
@@ -2,12 +2,93 @@
...
+The target audience of this document is the sbt plugin authors
+and sbt developers. See also [How can I help?][HowCanIHelp]
+
+### Modularization
+
+The process we aim to take for sbt 1.0 is to disassemble sbt into small modules.
```
Comment (jsuereth) [inline src/reference/03-Developers-Guide/00.md]: This also appears to be focused on the path to 1.0.   It really should be ina  "path to 1.0" section.

**Comment by jsuereth** [comment]
This looks like a 1.0 roadmap more than a developer guide.  It's still useful, but the developer guide really should teach you the following:

1. The layout of the code
2. The key conepts/abstractions you need to understand
3. What each component is meant to do.

What you wrote is very great material and needs to exist, but I think it's more of a "Pathway to 1.0" or "Upcoming Architectural Work".  If you read the content, it's very heavy on "We will being doing XYZ".

The notes from before, AFAIK, are relatively unchanged (caveat minor details) on the inner workings of sbt.   I think they still form the basis of a dev guide.  If we want to talk about sbt modules, we should start by discussing NOT sbt 1.0, but *just* the principles behind modularization and the current sbt modules that exist (even if they are in the github.com/sbt/sbt website).

The section you relegate to "notes" should still exist as explanations of key abstractions in sbt.

A dev guide is meant to get a developer up-to-speed on the code base.   A small part of that would be to outline current work.   I.e. the Dev Guide should help plugin authors as much as sbt-core authors.

File: src/reference/03-Developers-Guide/00.md (hunk 2 above)
Comment (eed3si9n) [inline src/reference/03-Developers-Guide/00.md]: Modularization <: Architecture. I want this section to be documentation of public facing API + guided tour of our code base eventually.
The fact that the whole thing is modular is *a* part of architecture IMHO as opposed to monolithic.

File: src/reference/03-Developers-Guide/00.md [hunk 4]
```diff
@@ -2,12 +2,93 @@
...
+programs resolved using the Ivy dependency manager. Sbt uses this as its
+own deployment mechanism. (Hosted at [sbt/launcher](https://github.com/sbt/launcher))
+
+#### [sbt/website](https://github.com/sbt/website)
+
+This website's source.
+
+#### [sbt/sbt](https://github.com/sbt/sbt)
+
+Most of the sbt codebase still lives here.
```
Comment (jsuereth) [inline src/reference/03-Developers-Guide/00.md]: We should list out the libraries in here which we plan to modularize:

* sbt/io
* sbt/log
* sbt/completion
* sbt/incremental-compiler (or sbt/zinc)
* sbt/collections

File: src/reference/03-Developers-Guide/00.md [hunk 5]
```diff
@@ -2,12 +2,93 @@
...
+Here's what we envision the modilarization process to look like:
+
+- pull apart subprojects by feature
+- make separate Github repositories
+- distinguish public API and internal implemention
+- document the usages
+- clean up the code since we can break bincompat
+- publish to Maven
+
+Here are some of the modules.
```
Comment (jsuereth) [inline src/reference/03-Developers-Guide/00.md]: We should list `sbt-remote-control` here.

File: src/reference/03-Developers-Guide/00.md [hunk 6]
```diff
@@ -2,12 +2,93 @@
...
+The only thing that will change is the last point.
+In sbt 1.0, we will replace the interactive shell with sbt server
+that's accessible via JSON API and a text-based client.
+
+#### Source compatibility of the build deinition
+
+Source compatibility means that a build source that worked for sbt version A
+works for another version B without modification.
+For example, a build defition created for sbt 1.0.0 should work also for sbt 1.1.0
+(if we adopted Semantic Versioning).
```
Comment (jsuereth) [inline src/reference/03-Developers-Guide/00.md]: s/if/when/

File: src/reference/03-Developers-Guide/00.md [hunk 7]
```diff
@@ -2,12 +2,93 @@
...
+1. incremental compilation that supports Scala
+2. dependency management that's aware of Scala's binary compatibility
+3. task and plugins system that's extensible using Scala
+4. a text-based interactive shell
+
+The only thing that will change is the last point.
+In sbt 1.0, we will replace the interactive shell with sbt server
+that's accessible via JSON API and a text-based client.
+
+#### Source compatibility of the build deinition
```
Comment (jsuereth) [inline src/reference/03-Developers-Guide/00.md]: Add a notice on "compatibility".   i.e. that we aim to keep compatibility between releases.  Some kind of mantra:

"Never break plugins, Never break build source files."

File: src/reference/03-Developers-Guide/00.md (hunk 4 above)
Comment (eed3si9n) [inline src/reference/03-Developers-Guide/00.md]: Done

File: src/reference/03-Developers-Guide/00.md (hunk 5 above)
Comment (eed3si9n) [inline src/reference/03-Developers-Guide/00.md]: Done

File: src/reference/03-Developers-Guide/00.md (hunk 7 above)
Comment (eed3si9n) [inline src/reference/03-Developers-Guide/00.md]: Done

**Comment by jsuereth** [comment]
LGTM

**Pull request merged**